app/controllers/hosts_controller.rb

```diff
@@ -137,6 +137,15 @@ def compute_resource_selected
     end
   end
 
+  def scheduler_hint_selected
+    return not_found unless (params[:host])
+    @host = Host.new params[:host]
+    Taxonomy.as_taxonomy @organization, @location do
```

Comment:
After making the change to `find_resource` you can also do without `Taxonomy.as_taxonomy`  `@host` and `compute_resource` are already visible to the current scope if you can find them